spec/safe_statements_spec.rb

```diff
@@ -2110,6 +2110,762 @@ def up
           end
         end
 
+        describe "safe_partman_create_parent" do
+          it "raises error when retention is set" do
+            migration = Class.new(migration_klass).new
+
+            expect(migration).to_not receive(:unsafe_partman_create_parent)
+
+            expect do
+              migration.safe_partman_create_parent(:foos3, retention: "60 days")
+            end.to raise_error(
+              PgHaMigrations::UnsafeMigrationError,
+              /:retention and\/or :retention_keep_table => false can potentially result in data loss if misconfigured/
+            )
+          end
+
+          it "raises error when retention_keep_table is set" do
+            migration = Class.new(migration_klass).new
+
+            expect(migration).to_not receive(:unsafe_partman_create_parent)
+
+            expect do
+              migration.safe_partman_create_parent(:foos3, retention_keep_table: false)
+            end.to raise_error(
+              PgHaMigrations::UnsafeMigrationError,
+              /:retention and\/or :retention_keep_table => false can potentially result in data loss if misconfigured/
+            )
+          end
+
+          it "delegates to unsafe_partman_create_parent when potentially dangerous args are not set" do
+            migration = Class.new(migration_klass).new
+
+            expect(migration).to receive(:unsafe_partman_create_parent).with(:foos3, arg1: "foo", arg2: "bar")
+
+            migration.safe_partman_create_parent(:foos3, arg1: "foo", arg2: "bar")
+          end
+        end
+
+        describe "unsafe_partman_create_parent" do
+          context "when extension not installed" do
+            it "raises error" do
+              migration = Class.new(migration_klass) do
+                def up
+                  safe_create_partitioned_table :foos3, type: :range, key: :created_at do |t|
```

Comment:
It might be nice to extract this to a `before(:each)`, which will also separate it from the actual code under test.

Reply:
I do think that makes sense, but it would require a pretty significant refactor to remove stuff like this from the body of _all_ tests (since there are some negative tests that rely on the parent table _not_ being present)

Reply:
Hmm, well, my bigger point is that having the code under test not be mixed with the code not under test is meaningful with these block based expectations.